### ISSUE TITLE
'Set browser as default' dialog on 2nd and nth launch

### ISSUE THREAD

**Issue by karenkliu** [brave/brave-browser#12203]
## Description 
One of the issues made evident from lapsed user research is that we can do better in encouraging users to set Brave as their default browser.

Currently, the "set browser dialog" is displayed as a non-user-blocking infobar on 2nd launch. 

## Designs
Change the infobar to a user-blocking browser dialog to encourage interaction.

![image](https://user-images.githubusercontent.com/2388823/96523566-dd8f1400-122a-11eb-8c82-146e6f3fdd4f.png)

Brave will check if it has been set as the default browser on every launch. This dialog reappears:

- If the user hits "not now", the dialog reappears on the 3rd, 4th, 8th, and 20th launch.
- If the user had set Brave as the default, and then it was not the default on next launch (possibly due to updating OS), then show the dialog regardless of the nth launch

Dark theme is supported:
![image](https://user-images.githubusercontent.com/2388823/104251351-5fef3480-5424-11eb-928a-7392dbab86dd.png)

## Assets
Figma: https://www.figma.com/file/MFHpcXJMg2RUQpUJMAZZ4R/?node-id=146%3A0

## Note
This is a follow up issue to:

- MacOS first launch: https://github.com/brave/brave-browser/issues/12679
- Windows first launch: https://github.com/brave/brave-browser/issues/12680

**Comment by karenkliu** [comment]
@rebron When do you want this to dialog to show again if the user selects "not now" ?

**Comment by karenkliu** [comment]
@rebron @bsclifton For this issue, we need to finish defining the reappearance of the dialog. What are your thoughts on this cadence (and if it's feasible):

>If the user hits "not now", the dialog reappears on the 3rd, 4th, 8th, and 20th launch.
If the user had set Brave as the default, and then it was not the default on next launch (possibly due to updating OS), then show the dialog regardless of the nth launch

**Comment by bsclifton** [comment]
I like the frequency - I think it makes sense to not reset the count if browser was set as default and then was removed as default. Basically the frequency is only incremented if you launch Brave + it's not default

**Comment by karenkliu** [comment]
Check that when we implement this issue it also fixes https://github.com/brave/brave-browser/issues/9306

**Comment by simonhong** [comment]
In chromium, dismissed infobar is re-appearred after some days passed. `period_days`.
In our case, fetched `period_days` is zero. So, maybe we don't show again after it's dismissed first time.
@iefremov Do you know we configure this variation value? 
```
  if (last_dismissed_value) {
    int period_days = 0;
    base::StringToInt(variations::GetVariationParamValue(
                          "DefaultBrowserInfobar", "RefreshPeriodDays"),
                      &period_days);
    if (period_days <= 0 || period_days == std::numeric_limits<int>::max())
      return false;  // Failed to parse a reasonable period.
    base::Time show_on_or_after =
        base::Time::FromInternalValue(last_dismissed_value) +
        base::TimeDelta::FromDays(period_days);
    if (base::Time::Now() < show_on_or_after)
      return false;
  }
```

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.21.56 Chromium: 88.0.4324.152 (Official Build) dev (64-bit)
-- | --
Revision | 6579930fc53b4dc589c042bec9d0a3778326974d-refs/branch-heads/4324@{#2106}
OS | Ubuntu 18.04 LTS

Verified test plan from https://github.com/brave/brave-core/pull/7716

Verified default browser dialog is shown after 2nd, 3rd, 4th, 8th and 20th launch. Verified `brave->default_browser->launching_count` in `Preferences` is increased each browser launch
![image](https://user-images.githubusercontent.com/34715963/108063356-5f693f80-705b-11eb-9a62-4a87707dc791.png)

Verified dialog is no longer shown after the browser is made default.

Verified dialog is shown after another browser is made default. Verified `brave->default_browser->launching_count` in `Preferences` starts from 1.

Verified dialog is not shown when `brave://welcome page` is active tab.

Verified on upgrade profile (1.20.x) that default browser dialog is shown.

Logged for missing translations

**Comment by btlechowski** [comment]
Set as `QA/Blocked` till we get a `release` channel build as the issue needs to be tested on all channels.

**Comment by LaurenWags** [comment]
Removing `QA/Blocked` since https://github.com/brave/brave-browser/releases/tag/v1.21.68 is available

**Comment by stephendonner** [comment]
Verified FIXED on `release` build, `macOS 11.2.2 (Big Sur)` using the testplan from https://github.com/brave/brave-core/pull/7716 on:



| Brave    | 1.21.68 Chromium: 89.0.4389.58 (Official Build) (x86_64)     |
| -------- | ------------------------------------------------------------ |
| Revision | 1a139f28ecc27719439e37c6b1533cee999cb802-refs/branch-heads/4389@{#1134} |
| OS       | macOS Version 11.2.2 (Build 20D80)                           |

**Verified:**

* both dark and light themes (only light pictured, sorry)
* default-browser dialog appears on the 2nd, 3rd, 4th, 8th, and 20th launches
* the above dialog does *NOT* appear on launches 9-19
* after choosing "Set as default", in the Brave dialog, then confirming with the OS-level dialog, the user is no longer prompted to set their default browser, on restart
* after the 20th launch, regardless of choosing `Not Now` or `Set as default` the modal dialog did *NOT* appear 

**Launches:**

| 2nd                                                          | 3rd                                                          | 4th                                                          | 8th                                                          | 20th                                                         |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
| <img src="https://user-images.githubusercontent.com/387249/109230209-56721f80-7779-11eb-8641-a557d25731eb.png"> | <img src="https://user-images.githubusercontent.com/387249/109230218-583be300-7779-11eb-9c6e-8a77d70d0c3c.png"> | <img src="https://user-images.githubusercontent.com/387249/109230217-583be300-7779-11eb-8187-2f6629fb725e.png"> | <img src="https://user-images.githubusercontent.com/387249/109230215-57a34c80-7779-11eb-87f9-086e260f389d.png"> | <img src="https://user-images.githubusercontent.com/387249/109230213-57a34c80-7779-11eb-8756-37164469115c.png"> |
